tools/tertiary-analysis/garnett/garnett_train_classifier.xml

```diff
@@ -5,12 +5,13 @@
     </macros>
     <expand macro="requirements" />
     <command detect_errors="exit_code"><![CDATA[
-        garnett_train_classifier.R --cds-object '${cds_object}' --marker-file-path '${marker_file_path}' --database '${database}' --cds-gene-id-type '${cds_gene_id_type}' --marker-file-gene-id-type '${marker_file_gene_id_type}' --num-unknown '${num_unknown}' --min-observations '${min_observations}' --max-training-samples '${max_training_samples}' --propogate-markers '${propogate_markers}' --cores '${cores}' --classifier-gene-id-type '${classifier_gene_id_type}' --output-path '${output_classifier_path}'
+        garnett_train_classifier.R --cds-object '${cds_object}' --marker-file-path '${marker_file_path}' --database '${database}' --train-idf '${train_idf}' --cds-gene-id-type '${cds_gene_id_type}' --marker-file-gene-id-type '${marker_file_gene_id_type}' --num-unknown '${num_unknown}' --min-observations '${min_observations}' --max-training-samples '${max_training_samples}' --propogate-markers '${propogate_markers}' --cores '${cores}' --classifier-gene-id-type '${classifier_gene_id_type}' --output-path '${output_classifier_path}'
```

Comment:
Are all the inputs required? Otherwise the command line should be constructed accordingly (with cheetah flow control statements).

Reply:
thanks for your feedback @pcm32. by cheetah statements, do you mean something like this?
```
#if $param != ''
--param $param 
#end if
```